### PR TITLE
chore(docs): updates for new falcoctl driver tool.

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

/area documentation

**What this PR does / why we need it**:

Updated falco-driver-loader related documentation to use new `falcoctl driver` tool that will be shipped by Falco 0.37.
Moreover, it updates the way to select a given driver in Falco, given the new `engine` config file options.

Finally, it uses a more coherent `legacy eBPF` naming in place of `traditional` or `classic`, everywhere.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by FedeDP** [comment]
Then, we will need a subsequent PR to also update ways to run Falco with different drivers, now that Falco deprecated old `FALCO_BPF_PROBE` and `--modern-bpf`, and make use of new `engine.kind` configuration key.

**Comment by FedeDP** [comment]
Ok, ported `FALCO_BPF_PROBE` and `--modern-bpf` documentation to new `engine.kind` Falco config option.
Moreover, dropped userspace instrumentation since it was already deprecated and its support has been removed from Falco in the 0.37 release cycle.

File: content/en/docs/event-sources/kernel/_index.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ The classic {{< glossary_tooltip text="eBPF probe" term_id="ebpf-probe" >}} is a
 
 To install the eBPF probe, please refer to the [installation](/docs/getting-started/installation/#install-driver) page.
 
-To enable the eBPF support in Falco set the `FALCO_BPF_PROBE` environment variable to an empty value (ie. `FALCO_BPF_PROBE=""`, in this way Falco will search under `$HOME/.falco` directory for a file called `falco-bpf.o`) or otherwise explicitly set it to the path where the eBPF probe resides.
+To enable the eBPF support in Falco set the `engine.kind` configuration key to `ebpf` and eventually customize `engine.ebpf.probe`  to the path where the eBPF probe resides; the default path is the location used by `falcoctl driver` tool to install the eBPF probe.
```
Comment (LucaGuerra) [inline content/en/docs/event-sources/kernel/_index.md:39]: I have a suggestion for this: why don't we move the modern eBPF probe section up here, update the instructions to run it and move the classic/legacy one down? This way people with recent kernels will read relevant instruction first and have a better experience and only people that need eBPF but cannot run the new probe will follow the legacy instructions to run Falco on their system

Comment (FedeDP) [reply content/en/docs/event-sources/kernel/_index.md:39]: It's a great idea actually!
cc @leogr @Andreagit97 @incertum wdyt?

Comment (leogr) [reply content/en/docs/event-sources/kernel/_index.md:39]: totally agree

Comment (FedeDP) [reply content/en/docs/event-sources/kernel/_index.md]: Done! Let me know wdyt!

**Comment by FedeDP** [comment]
Rebased on top of master.

File: content/en/docs/event-sources/kernel/_index.md [hunk 2]
```diff
@@ -12,8 +12,8 @@ Falco uses different instrumentations to analyze the system workload and pass se
 There are several supported drivers:
 
 - Kernel module *(default)*
-- Classic eBPF probe
 - Modern eBPF probe
+- Classic eBPF probe
```
Comment (incertum) [inline content/en/docs/event-sources/kernel/_index.md:16]: `Classic` or didn't we agree on `Legacy`? My preference would be `Legacy` as I believe most other docs call it that way.

File: content/en/docs/event-sources/kernel/_index.md [hunk 3]
```diff
@@ -22,57 +22,14 @@ There are several supported drivers:
...
-		return sysctl_perf_event_paranoid > 2;
-	}
-	```
-
- 	As you can notice, when your `kernel.perf_event_paranoid` is `>2` the capability `CAP_PERFMON` won't suffice, you would still need `CAP_SYS_ADMIN`.
- 	So before disabling `CAP_SYS_ADMIN` check your `perf_event_paranoid` value with `sysctl kernel.perf_event_paranoid` and make sure their values are compatible with your distribution enforcement.
-
 ## Modern eBPF probe
 
 The {{< glossary_tooltip text="modern probe" term_id="modern-ebpf-probe" >}} is an alternative driver for Falco. The main advantage it brings to the table is that it is embedded into Falco, which means that you don't have to download or build anything, if your kernel is recent enough Falco will automatically inject it!
```
Comment (incertum) [inline content/en/docs/event-sources/kernel/_index.md:35]: modern probe? stay consistent with modern_ebpf?

File: content/en/docs/event-sources/kernel/_index.md [hunk 4]
```diff
@@ -97,13 +54,16 @@ sudo bpftool feature probe kernel | grep -q "program_type tracing is available"
 
 ### How to run it
 
-The modern eBPF probe supports all the installation methods of other drivers:
+To enable the modern eBPF support in Falco, just set the `engine.kind` configuration key to `modern_ebpf`. Nothing else will be needed since no external artifact is required for it to work.
```
Comment (incertum) [inline content/en/docs/event-sources/kernel/_index.md:57]: Perhaps say something similar to https://github.com/falcosecurity/falco-website/pull/1246/files#diff-c3777e5fb1d24e6ce0aae8a99b12234f648c9ef9ed17e4b858fce246deffad5eR40

File: content/en/docs/event-sources/kernel/_index.md [hunk 5]
```diff
@@ -123,33 +83,45 @@ The mentioned capabilities require no further explanation since they were alread
...
-{{% pageinfo color="warning" %}}
-The userspace instrumentation is deprecated and it will be removed in Falco `0.37.0`.
-{{% /pageinfo %}}
+The classic {{< glossary_tooltip text="eBPF probe" term_id="ebpf-probe" >}} is an alternative source to the ones described above, leveraging greater compatibility than the modern BPF one, since it requires older kernel versions.
 
-Differently from the other drivers, as the name suggests, userspace instrumentation happens 100% in userspace.
+To install the eBPF probe, please refer to the [installation](/docs/getting-started/installation/#install-driver) page.
 
-The Falco community, in 0.24.0 promoted the userspace instrumentation feature to **[official support](https://github.com/falcosecurity/evolution#official-support)** to be included in Falco.
+To enable the eBPF support in Falco set the `engine.kind` configuration key to `ebpf` and eventually customize `engine.ebpf.probe`  to the path where the eBPF probe resides; the default path is the location used by `falcoctl driver` tool to install the eBPF probe.
```
Comment (incertum) [inline content/en/docs/event-sources/kernel/_index.md:92]: Tell what the default path is here as well.

File: content/en/docs/install-operate/download.md [hunk 6]
```diff
@@ -36,7 +36,7 @@ If you use Falco for non-syscall events, such as some plugins, only download the
...
+2. Falco with kernel module or traditional BPF driver: Also, download the Falco binary artifact, and additionally, download the kernel artifact corresponding to your kernel release (`uname -r`) for either `.ko` (kernel module) or `.o` (BPF driver) from the [driver](https://download.falco.org/?prefix=driver/) store. Navigate to the driver versions' directory that is compatible with the Falco binary (check with `falco --version`). To make this easier, Falco has a `falcoctl driver` component that automates the driver download or tries to build it. The [Install](/docs/install-operate/installation/) guide will explain this more and the text blob below also has more information.
```
Comment (incertum) [inline content/en/docs/install-operate/download.md:39]: And we call it traditional BPF driver -> change to consistent `legacy eBPF` driver

Comment (incertum) [reply content/en/docs/install-operate/download.md:39]: Also change all old BPF driver references to eBPF ...

File: content/en/docs/install-operate/upgrade.md [hunk 7]
```diff
@@ -109,6 +109,6 @@ For the Falco binary we don't provide specific update paths, you just have to re
...
 
-When performing kernel upgrades on your host, a reboot is required. Consequently, the Falco binary restarts, and additionally, you must ensure that a new kernel driver corresponding to the updated kernel release (`uname -r`) is available when using the kernel module or traditional BPF driver. By using Falco's `falco-driver-loader`, these processes are automated for you, making it easy to handle kernel upgrades. The Falco Project features a kernel crawler and automated CI, ensuring you can always obtain the necessary pre-built driver artifact, even for the latest kernel releases we support.
+When performing kernel upgrades on your host, a reboot is required. Consequently, the Falco binary restarts, and additionally, you must ensure that a new kernel driver corresponding to the updated kernel release (`uname -r`) is available when using the kernel module or traditional BPF driver. By using Falco's `falcoctl driver` tool, these processes are automated for you, making it easy to handle kernel upgrades. The Falco Project features a kernel crawler and automated CI, ensuring you can always obtain the necessary pre-built driver artifact, even for the latest kernel releases we support.
```
Comment (incertum) [inline content/en/docs/install-operate/upgrade.md:112]: same old "traditional BPF driver" ref

File: content/en/docs/install-operate/upgrade.md [hunk 8]
```diff
@@ -109,6 +109,6 @@ For the Falco binary we don't provide specific update paths, you just have to re
...
 
 The great news is that modern BPF driver is more resilient to it, because of the CO-RE "Compile Once - Run Everywhere" feature that made it possible to bundle the driver into the Falco binary - it will just continue to work on the upgraded kernel. If possible, use modern BPF!
```
Comment (incertum) [inline content/en/docs/install-operate/upgrade.md:114]: Actually could we string search for "modern BPF" and "traditional BPF" or "classic BPF" and change everything in one swing?

File: content/en/docs/install-operate/download.md (hunk 6 above)
Comment (FedeDP) [reply content/en/docs/install-operate/download.md:39]: Great catches!

**Comment by FedeDP** [comment]
@incertum done everything! Thanks for the huge feedback!

**Review by leogr — APPROVED** [review]
LGTM

/hold for the 0.37

**Pull request merged**